### ISSUE TITLE
[ERROR] Failed at command: /opt/homebrew/bin/brew info --json=v1 --eval-all Error: formulae require at least a URL

### ISSUE THREAD

**Issue by Drew-Daniels** [rcmdnk/homebrew-file#282]
I'm getting an error on a fresh install of `homebrew-file` after installing the formula, and updating my `zsh` and `fish` configs to use `brew-wrap`:

My `XDG_CONFIG_HOME` variable is set to `~/projects/dotfiles` in my `.zshrc`:

I have installed the formula:
```sh
$ brew install rcmdnk/file/brew-file
```

Attempted running `brew init` - following either the `y` or `n` paths both ultimately yield the same error message when `brew-wrap` tries generating a `Brewfile`:
```sh
$ brew init
Do you want to set a repository (y)? ((n) for local Brewfile). [y/n]: y

Set repository,
"non" (or empty) for local Brewfile (/Users/drew/projects/dotfiles/brewfile/Brewfile),
/path/to/repo for local git repository,
https://your/git/repository (or ssh://user@server.project.git) for git repository,
or (<user>/)<repo> for github repository,
or full path for other git repository: non

################################
# Set Brewfile repository as non
################################


############################################################
# Initialize /Users/drew/projects/dotfiles/brewfile/Brewfile
############################################################

[ERROR] Failed at command: /opt/homebrew/bin/brew info --json=v1 --eval-all
Error: formulae require at least a URL
/opt/homebrew/Library/Homebrew/formula.rb:319:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:253:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
/opt/homebrew/Library/Homebrew/formula.rb:2139:in `block in all'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `each'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `filter_map'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `all'
/opt/homebrew/Library/Homebrew/cmd/info.rb:201:in `print_json'
/opt/homebrew/Library/Homebrew/cmd/info.rb:98:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
Please report this issue:
  https://docs.brew.sh/Troubleshooting
```

`homebrew`-related configuration in `.zshrc`:

```sh
# homebrew
PATH="/usr/local/sbin:$PATH"

eval "$(/opt/homebrew/bin/brew shellenv)"

# homebrew shell completion
if type brew &>/dev/null; then
  FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"
fi

# brew curl shell completion
fpath=("$(brew --prefix)/opt/curl/share/zsh/site-functions" $fpath)

# https://homebrew-file.readthedocs.io/en/latest/installation.html
if [ -f $(brew --prefix)/etc/brew-wrap ]; then
  source $(brew --prefix)/etc/brew-wrap
fi

export HOMEBREW_BUNDLE_NO_LOCK=1
```

`homebrew`-related configuration in `fish` config:
```fish
  # homebrew
  fish_add_path /usr/local/sbin
  set -gx HOMEBREW_BUNDLE_NO_LOCK 1

  # https://homebrew-file.readthedocs.io/en/latest/installation.html
  if test -f (brew --prefix)/etc/brew-wrap.fish
      source (brew --prefix)/etc/brew-wrap.fish
  end
```

```sh
$ uname -a
Darwin drews-m1-mbp 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000 arm64
```

**Comment by rcmdnk** [comment]
It could be related with issues such as:
https://github.com/orgs/Homebrew/discussions/879
https://github.com/Homebrew/brew/issues/16123

Could you please give me `brew ls` results?
or try:

```
for f in $(brew ls --formula);do if ! json=$(brew info --json=v1 $f 2>&1);then echo $f;echo $json;fi;done
```

It may takes a time, but will show which formula makes the error.

**Comment by rcmdnk** [comment]
Could you give me `brew ls` results?

Not sure what is different in the command before and after installing `homebrew-file`...

To check which formula is wrong after installing `homebrew-file`, 
you may try (show all formula, and show result only when it fails):

```
$ for f in $(brew ls --formula);do echo $f;if ! json=$(brew info --json=v1 $f 2>&1);then echo $json;fi;done
```

**Comment by Drew-Daniels** [comment]
It looks like `brew ls` is failing after I install `homebrew-file` and it never returns:
```sh
$ brew ls
# gets stuck
```

Also, FWIW, when I uninstall `homebrew-file`, I see the same error message:

```sh
❯ brew uninstall rcmdnk/file/brew-file
Uninstalling /opt/homebrew/Cellar/brew-file/9.0.16... (11 files, 189.3KB)
[ERROR] Failed at command: /opt/homebrew/bin/brew info --json=v1 --eval-all
Error: formulae require at least a URL
/opt/homebrew/Library/Homebrew/formula.rb:319:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:253:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
/opt/homebrew/Library/Homebrew/formula.rb:2139:in `block in all'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `each'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `filter_map'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `all'
/opt/homebrew/Library/Homebrew/cmd/info.rb:201:in `print_json'
/opt/homebrew/Library/Homebrew/cmd/info.rb:98:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
Please report this issue:
  https://docs.brew.sh/Troubleshooting

```

Once uninstalled, I see all of the results as expected:
```sh
❯ brew ls -1
==> Formulae
actionlint
aom
aribb24
autoconf
automake
bat
bat-extras
bottom
brotli
bun
ca-certificates
cairo
chafa
chezmoi
cjson
cmake
cmake-docs
comby
coreutils
ctags
curl
dav1d
dust
expat
fd
ffmpeg
fish
flac
fontconfig
freetype
frei0r
fribidi
fzf
gdk-pixbuf
gettext
gh
ghostscript
giflib
git
git-gui
glib
gmp
gnutls
gping
graphite2
harfbuzz
highway
hurl
icu4c
imagemagick
imath
jasper
jbig2dec
jira-cli
jless
jpeg-turbo
jpeg-xl
jq
lame
leiningen
leptonica
libapplewm
libarchive
libass
libb2
libbluray
libde265
libev
libevent
libfontenc
libgit2
libgit2@1.7
libheif
libice
libidn
libidn2
liblqr
libmicrohttpd
libnghttp2
libogg
libomp
libpng
libraw
librist
librsvg
libsamplerate
libsm
libsndfile
libsodium
libsoxr
libssh
libssh2
libtasn1
libtiff
libtool
libunibreak
libunistring
libuv
libvidstab
libvmaf
libvorbis
libvpx
libvterm
libx11
libxau
libxcb
libxdmcp
libxext
libxfixes
libxfont2
libxi
libxinerama
libxkbcommon
libxkbfile
libxmu
libxrandr
libxrender
libxt
libxtst
libyaml
little-cms2
llvm
lpeg
lsd
luajit
luv
lz4
lzo
m4
mas
mbedtls
mesa
mise
mpdecimal
mpg123
msgpack
ncurses
neovim
nettle
oniguruma
opencore-amr
openexr
openjdk
openjpeg
openssl@3
opus
p11-kit
pango
pcre
pcre2
pixman
pkg-config
python@3.12
rav1e
readline
reattach-to-user-namespace
ripgrep
rtmpdump
rubberband
rust
sdl2
shared-mime-info
snappy
speedtest
speex
sqlite
srt
starship
stylua
svt-av1
tcl-tk
terraform
tesseract
theora
tmux
tree
tree-sitter
unbound
unibilium
usage
utf8proc
webp
wget
x264
x265
xauth
xcb-util
xcb-util-image
xcb-util-keysyms
xcb-util-renderutil
xcb-util-wm
xdotool
xkbcomp
xkeyboardconfig
xorg-server
xorgproto
xvid
xxhash
xz
yazi
z3
zeromq
zimg
zoxide
zstd

==> Casks
alex313031-thorium
alt-tab
dbeaver-community
discord
docker
drawio
firefox
font-jetbrains-mono-nerd-font
google-chrome
jellyfin
obs
postgres-unofficial
postman
qbittorrent
raindropio
raspberry-pi-imager
rectangle
roboform
signal
slack
tidal
tor-browser
vitalsource-bookshelf
zoom
```

When I run `for f in $(brew ls --formula);do echo $f;if ! json=$(brew info --json=v1 $f 2>&1);then echo $json;fi;done` with `homebrew-file` it never prints anything (I'm guessing because `brew ls` never returns anything in this case), but it does successfully print out all formulae without error messages once `homebrew-file` is uninstalled:

```sh
❯ for f in $(brew ls --formula);do echo $f;if ! json=$(brew info --json=v1 $f 2>&1);then echo $json;fi;done
actionlint
aom
aribb24
autoconf
automake
bat
bat-extras
bottom
brotli
bun
ca-certificates
cairo
chafa
chezmoi
cjson
cmake
cmake-docs
comby
coreutils
ctags
curl
dav1d
dust
expat
fd
ffmpeg
fish
flac
fontconfig
freetype
frei0r
fribidi
fzf
gdk-pixbuf
gettext
gh
ghostscript
giflib
git
git-gui
glib
gmp
gnutls
gping
graphite2
harfbuzz
highway
hurl
icu4c
imagemagick
imath
jasper
jbig2dec
jira-cli
jless
jpeg-turbo
jpeg-xl
jq
lame
leiningen
leptonica
libapplewm
libarchive
libass
libb2
libbluray
libde265
libev
libevent
libfontenc
libgit2
libgit2@1.7
libheif
libice
libidn
libidn2
liblqr
libmicrohttpd
libnghttp2
libogg
libomp
libpng
libraw
librist
librsvg
libsamplerate
libsm
libsndfile
libsodium
libsoxr
libssh
libssh2
libtasn1
libtiff
libtool
libunibreak
libunistring
libuv
libvidstab
libvmaf
libvorbis
libvpx
libvterm
libx11
libxau
libxcb
libxdmcp
libxext
libxfixes
libxfont2
libxi
libxinerama
libxkbcommon
libxkbfile
libxmu
libxrandr
libxrender
libxt
libxtst
libyaml
little-cms2
llvm
lpeg
lsd
luajit
luv
lz4
lzo
m4
mas
mbedtls
mesa
mise
mpdecimal
mpg123
msgpack
ncurses
neovim
nettle
oniguruma
opencore-amr
openexr
openjdk
openjpeg
openssl@3
opus
p11-kit
pango
pcre
pcre2
pixman
pkg-config
python@3.12
rav1e
readline
reattach-to-user-namespace
ripgrep
rtmpdump
rubberband
rust
sdl2
shared-mime-info
snappy
speedtest
speex
sqlite
srt
starship
stylua
svt-av1
tcl-tk
terraform
tesseract
theora
tmux
tree
tree-sitter
unbound
unibilium
usage
utf8proc
webp
wget
x264
x265
xauth
xcb-util
xcb-util-image
xcb-util-keysyms
xcb-util-renderutil
xcb-util-wm
xdotool
xkbcomp
xkeyboardconfig
xorg-server
xorgproto
xvid
xxhash
xz
yazi
z3
zeromq
zimg
zoxide
zstd
```

Thanks again for looking into this. Seems like wherever things are erroring out it's a little hard to decipher.

**Comment by Drew-Daniels** [comment]
In case it's helpful:

```sh
❯ brew config
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 9 hours ago
Core tap JSON: 02 Sep 18:07 UTC
Core cask tap JSON: 02 Sep 18:07 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.KH0B7Q6W3P/org.xquartz:0
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.46.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false

❯ brew doctor
Your system is ready to brew.

```

**Comment by rcmdnk** [comment]
there could be some conflicts with brew-wrap.
(but it seems not related with `brew info`, another error...?)

Could you please try after installing homebrew-file (and set up brew-wrap):

```
$ type brew # check if wrapped or not
$ type _post_brewfile_update # This should be empty function, if you do not implement
$ command brew ls # This should be fine
$ brew ls # this does not work?
$ brew-file cat # One of commands executed in brew wrapped by brew-wrap. It shows contents of Brewfile
```

**Comment by rcmdnk** [comment]
If possible, could you please try bash?

```
$ bash
$ source <path/to/brew-wrap>
$ brew ls
$ brew init
```

**Comment by Drew-Daniels** [comment]
Sure, here you go. These are what I get when running in `zsh`, not `fish`:

```sh
❯ type brew
brew is a shell function from /opt/homebrew/etc/brew-wrap

❯ type _post_brewfile_update
_post_brewfile_update is a shell function from /opt/homebrew/etc/brew-wrap

❯ brew ls
# hangs forever

❯ brew-file cat
[WARNING] Input file /Users/drew/projects/dotfiles/brewfile/Brewfile is not found.
Do you want to initialize from installed packages? [y/n]:
```

When running `bash`, `brew ls` and `brew init` hang - even after configuring my `~/.bashrc` as shown [here](https://homebrew-file.readthedocs.io/en/latest/installation.html#installation):

```bash
# homebrew
if [ -f $(brew --prefix)/etc/brew-wrap ]; then
  source $(brew --prefix)/etc/brew-wrap
fi
```

Having a little trouble running `source <path/to/brew-wrap>` part (not literally), but it's because I can't find where `brew-wrap` is installed - since `brew --prefix` never exits with `homebrew-file` installed and configured:

```bash
❯ brew --prefix
# never exits

❯ source $(brew --prefix)/etc/brew-wrap
# never exits
```

**Comment by rcmdnk** [comment]
Thanks.

Regarding the issue where `brew ls` does not work after installing brew-file, there was a bug, which has been fixed in version 9.0.17.

(This is actually a significant bug, and I really appreciate your efforts in debugging it!)


I believe the `brew init` issue is a separate problem, but you may want to try the following command first after installing brew-file:

```
$ brew init
```

after installing brew-file.

If that doesn't resolve the issue, you can try running the following commands with brew-file installed:

```
$ for f in $(brew ls --formula);do echo $f;if ! json=$(brew info --json=v1 $f 2>&1);then echo $json;fi;done
```

If it works, try also:

```
$ brew info --json=v1 --eval-all
```

---
Bug Description:

The issue occurred when the Brewfile was not initialized. The brew command, wrapped by brew-wrap, executes brew file cat. If the Brewfile does not exist, it prompts the user with:

```
Do you want to initialize from installed packages? [y/n]:
```

However, the brew-wrap function hides this prompt, causing the process to hang.

As of version 9.0.17, if the Brewfile is not initialized, brew file cat will now display an error, and brew-wrap will simply ignore the result of the cat command.

**Comment by Drew-Daniels** [comment]
Sure thing, this plugin seems really useful so I'm happy to help.

After reinstalling the latest version of `homebrew-file` (`v9.0.17`), and running `brew init`, I still get the same error unfortunately. For extra context, I get the same error when either choosing:
- `n` to the "set repository" step
- `y` to the "set repository" step, and then typing "non"

```sh
❯ brew list --versions rcmdnk/file/brew-file
brew-file 9.0.17

❯ brew init
Do you want to set a repository (y)? ((n) for local Brewfile). [y/n]: y

Set repository,
"non" (or empty) for local Brewfile (/Users/drew/projects/dotfiles/brewfile/Brewfile),
/path/to/repo for local git repository,
https://your/git/repository (or ssh://user@server.project.git) for git repository,
or (<user>/)<repo> for github repository,
or full path for other git repository: non

################################
# Set Brewfile repository as non
################################


############################################################
# Initialize /Users/drew/projects/dotfiles/brewfile/Brewfile
############################################################

[ERROR] Failed at command: /opt/homebrew/bin/brew info --json=v1 --eval-all
Error: formulae require at least a URL
/opt/homebrew/Library/Homebrew/formula.rb:319:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:253:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
/opt/homebrew/Library/Homebrew/formula.rb:2139:in `block in all'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `each'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `filter_map'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `all'
/opt/homebrew/Library/Homebrew/cmd/info.rb:201:in `print_json'
/opt/homebrew/Library/Homebrew/cmd/info.rb:98:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
Please report this issue:
  https://docs.brew.sh/Troubleshooting
```

On the bright-side, looks like the latest version fixed the issue with `brew ls` for me, so here is the full output of running that shell script:

```sh
❯ for f in $(brew ls --formula);do echo $f;if ! json=$(brew info --json=v1 $f 2>&1);then echo $json;fi;done
actionlint
aom
aribb24
autoconf
automake
bat
bat-extras
bottom
brew-file
brotli
bun
ca-certificates
cairo
chafa
chezmoi
cjson
cmake
cmake-docs
comby
coreutils
ctags
curl
dav1d
dust
expat
fd
ffmpeg
fish
flac
fontconfig
freetype
frei0r
fribidi
fzf
gdk-pixbuf
gettext
gh
ghostscript
giflib
git
git-gui
glib
gmp
gnutls
gping
graphite2
harfbuzz
highway
hurl
icu4c
imagemagick
imath
jasper
jbig2dec
jira-cli
jless
jpeg-turbo
jpeg-xl
jq
lame
leiningen
leptonica
libapplewm
libarchive
libass
libb2
libbluray
libde265
libev
libevent
libfontenc
libgit2
libgit2@1.7
libheif
libice
libidn
libidn2
liblqr
libmicrohttpd
libnghttp2
libogg
libomp
libpng
libraw
librist
librsvg
libsamplerate
libsm
libsndfile
libsodium
libsoxr
libssh
libssh2
libtasn1
libtiff
libtool
libunibreak
libunistring
libuv
libvidstab
libvmaf
libvorbis
libvpx
libvterm
libx11
libxau
libxcb
libxdmcp
libxext
libxfixes
libxfont2
libxi
libxinerama
libxkbcommon
libxkbfile
libxmu
libxrandr
libxrender
libxt
libxtst
libyaml
little-cms2
llvm
lpeg
lsd
luajit
luv
lz4
lzo
m4
mas
mbedtls
mesa
mise
mpdecimal
mpg123
msgpack
ncurses
neovim
nettle
oniguruma
opencore-amr
openexr
openjdk
openjpeg
openssl@3
opus
p11-kit
pango
pcre
pcre2
pixman
pkg-config
python@3.12
rav1e
readline
reattach-to-user-namespace
ripgrep
rtmpdump
rubberband
rust
sdl2
shared-mime-info
snappy
speedtest
speex
sqlite
srt
starship
stylua
svt-av1
tcl-tk
terraform
tesseract
theora
tmux
tree
tree-sitter
unbound
unibilium
usage
utf8proc
webp
wget
x264
x265
xauth
xcb-util
xcb-util-image
xcb-util-keysyms
xcb-util-renderutil
xcb-util-wm
xdotool
xkbcomp
xkeyboardconfig
xorg-server
xorgproto
xvid
xxhash
xz
yazi
z3
zeromq
zimg
zoxide
zstd
```

Looks like this ends up displaying the same error as when I run `brew init`, so maybe there's something here:
```sh
❯ brew info --json=v1 --eval-all
Error: formulae require at least a URL
/opt/homebrew/Library/Homebrew/formula.rb:319:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:253:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
/opt/homebrew/Library/Homebrew/formula.rb:2139:in `block in all'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `each'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `filter_map'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `all'
/opt/homebrew/Library/Homebrew/cmd/info.rb:201:in `print_json'
/opt/homebrew/Library/Homebrew/cmd/info.rb:98:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
Please report this issue:
  https://docs.brew.sh/Troubleshooting
```

**Comment by rcmdnk** [comment]
these formula seem fine (all have url)
maybe some wrong file in taps?

Could you please give a list of taps?

```
$ brew tap
```

**Comment by rcmdnk** [comment]
try:

```
$ brew info --json=v1 --eval-all --debug
```

This will show something like below:


```
...
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/rcmdnk/homebrew-file/brew-file.rb
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/rcmdnk/homebrew-rcmdnkpac/chatgpt-prompt-wrapper.rb
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/rcmdnk/homebrew-rcmdnkpac/cocoro.rb
Error: formulae require at least a URL
Warning: Removed Sorbet lines from backtrace!
Rerun with `--verbose` to see the original backtrace
/opt/homebrew/Library/Homebrew/formula.rb:319:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:253:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
/opt/homebrew/Library/Homebrew/formula.rb:2139:in `block in all'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `each'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `filter_map'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `all'
/opt/homebrew/Library/Homebrew/cmd/info.rb:201:in `print_json'
/opt/homebrew/Library/Homebrew/cmd/info.rb:98:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
Please report this issue:
  https://docs.brew.sh/Troubleshooting
```

In this case, I commented out `URL` in formula `cocoro.rb`, which is not installed in the current machine.

You can find which formula has the problem by this command.

**Comment by Drew-Daniels** [comment]
Ok that's interesting, looks like it has something to do with the `hashicorp/tap` tap I have installed:

```sh
❯ brew tap
ankitpokhrel/jira-cli
hashicorp/tap
homebrew/bundle
homebrew/services
oven-sh/bun
rcmdnk/file
teamookla/speedtest

❯ brew ls -1
==> Formulae
actionlint
aom
aribb24
autoconf
automake
bat
bat-extras
bottom
brew-file
brotli
bun
ca-certificates
cairo
chafa
cjson
cmake
cmake-docs
comby
coreutils
ctags
curl
dav1d
dust
expat
fd
ffmpeg
fish
flac
fontconfig
freetype
frei0r
fribidi
fzf
gdk-pixbuf
gettext
gh
ghostscript
giflib
git
git-gui
glib
gmp
gnutls
gping
graphite2
harfbuzz
highway
hurl
icu4c
imagemagick
imath
jasper
jbig2dec
jira-cli
jless
jpeg-turbo
jpeg-xl
jq
lame
leiningen
leptonica
libapplewm
libarchive
libass
libb2
libbluray
libde265
libev
libevent
libfontenc
libgit2
libgit2@1.7
libheif
libice
libidn
libidn2
liblqr
libmicrohttpd
libnghttp2
libogg
libomp
libpng
libraw
librist
librsvg
libsamplerate
libsm
libsndfile
libsodium
libsoxr
libssh
libssh2
libtasn1
libtiff
libtool
libunibreak
libunistring
libuv
libvidstab
libvmaf
libvorbis
libvpx
libvterm
libx11
libxau
libxcb
libxdmcp
libxext
libxfixes
libxfont2
libxi
libxinerama
libxkbcommon
libxkbfile
libxmu
libxrandr
libxrender
libxt
libxtst
libyaml
little-cms2
llvm
lpeg
lsd
luajit
luv
lz4
lzo
m4
mas
mbedtls
mesa
mise
mpdecimal
mpg123
msgpack
ncurses
neovim
nettle
oniguruma
opencore-amr
openexr
openjdk
openjpeg
openssl@3
opus
p11-kit
pango
pcre
pcre2
pixman
pkg-config
pngpaste
python@3.12
rav1e
readline
reattach-to-user-namespace
ripgrep
rtmpdump
rubberband
rust
sdl2
shared-mime-info
snappy
speedtest
speex
sqlite
srt
starship
stylua
svt-av1
tcl-tk
terraform
tesseract
theora
tmux
tree
tree-sitter
unbound
unibilium
usage
utf8proc
webp
wget
x264
x265
xauth
xcb-util
xcb-util-image
xcb-util-keysyms
xcb-util-renderutil
xcb-util-wm
xdotool
xkbcomp
xkeyboardconfig
xorg-server
xorgproto
xvid
xxhash
xz
yazi
z3
zeromq
zimg
zoxide
zstd

==> Casks
alex313031-thorium
alt-tab
dbeaver-community
discord
docker
drawio
firefox
font-jetbrains-mono-nerd-font
google-chrome
jellyfin
obs
postgres-unofficial
postman
qbittorrent
raindropio
raspberry-pi-imager
rectangle
roboform
signal
slack
tidal
tor-browser
vitalsource-bookshelf
zoom
```

```sh
❯ brew info --json=v1 --eval-all --debug
# ... abbreviated
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/hashicorp/homebrew-tap/Formula/terraform.rb
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/hashicorp/homebrew-tap/Formula/tfstacks.rb
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /opt/homebrew/Library/Taps/hashicorp/homebrew-tap/Formula/vagrant.rb
Error: formulae require at least a URL
/opt/homebrew/Library/Homebrew/formula.rb:319:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:253:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:571:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:1016:in `factory'
/opt/homebrew/Library/Homebrew/formula.rb:2139:in `block in all'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `each'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `filter_map'
/opt/homebrew/Library/Homebrew/formula.rb:2138:in `all'
/opt/homebrew/Library/Homebrew/cmd/info.rb:201:in `print_json'
/opt/homebrew/Library/Homebrew/cmd/info.rb:98:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
Please report this issue:
  https://docs.brew.sh/Troubleshooting
```

What's more interesting is that `brew info --json=v1 --eval-all --debug` fails at `/opt/homebrew/Library/Taps/hashicorp/homebrew-tap/Formula/vagrant.rb` with and without `rcmdnk/file/brew-file` installed, so I guess this might be an unrelated error to `rcmdnk/file/brew-file` after all, that was only surfaced in my case whenever running commands from `homebrew-file`?

Looks like there is already an [outstanding issue](https://github.com/hashicorp/homebrew-tap/issues/258) in the Hashicorp tap repository about this.

I was able to confirm after uninstalling the `terraform` formula and untapping the `hashicorp/tap`, I was able to successfully run `brew info --json=v1 --eval-all --debug` and run `brew init` successfully and see it generate the `Brewfile` 🚀 

Thanks for all the help - and sorry this was unrelated to your formula!

**Comment by rcmdnk** [comment]
Thanks for the confirmation.

Though this should be fixed by the formula,
there could be people who need `hashicorp/tap` for some formulae, other than vagrant.

I will try to avoid the problem in #283.

This issue will keep opened at least until hashicorp tap fixes the problem or #283 is done.
